modules/api/pkg/handler/v2/external_cluster/kubeone.go

```diff
@@ -368,6 +370,23 @@ func setOpenStackCredentials(preset *kubermaticv1.Preset, kubeOneCloudSpec apiv2
 	return &kubeOneCloudSpec, nil
 }
 
+func setVsphereCredentials(preset *kubermaticv1.Preset, kubeOneCloudSpec apiv2.KubeOneCloudSpec) (*apiv2.KubeOneCloudSpec, error) {
+	if preset.Spec.VSphere == nil {
+		return nil, emptyCredentialError(preset.Name, "VSphere")
+	}
+
+	credentials := preset.Spec.VSphere
+
+	if kubeOneCloudSpec.VSphere == nil {
+		kubeOneCloudSpec.VSphere = &apiv2.KubeOneVSphereCloudSpec{}
+	}
+
+	kubeOneCloudSpec.VSphere.Username = credentials.Username
+	kubeOneCloudSpec.VSphere.Password = credentials.Password
```

Comment:
the thing is i think we need to update the struct for preset because now we use the same struct for kkp preset, and kubeone preset need to be different so right now the preset are not working fine in some of kubeone provides but i thought we can merge this now and fix the issue for the preset later since it need changes from the kkp repo also.
@ahmedwaleedmalik WDYT

Reply:
IMO if preset is not working then we should at least disable it for this provider. But this leads to inconsistent behaviour so I think we should fix the backend issue and merge all changes together.

Reply:
Agreed with @Waseem826, we can't ship a feature in this state. Since we are so close to code freeze, let's either fix the presets as well or skip shipping this in KKP 2.23.

Reply:
/hold